integration-azure-oauth-test/integration-azure-oauth-test-pageobjects/src/main/java/com/xwiki/entraid/test/po/EntraIDViewPage.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.entraid.test.po;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.xwiki.test.ui.po.ViewPage;
+
+/**
+ * Represents actions that can be done on the UI added by EntraID OIDC provider.
+ *
+ * @version $Id$
+ * @since 2.0
+ */
+public class EntraIDViewPage extends ViewPage
+{
+    @FindBy(xpath = "//a[@id='tmLogin']")
+    private WebElement loginLink;
+
+    @FindBy(xpath = "//a[@id='tmLogin-bypass']")
+    private WebElement bypassLogin;
+
+    @FindBy(id = "exceptionMessageContainer")
+    private WebElement microsoftContainer;
+
+    /**
+     * Go to XWiki Main WebHome.
+     */
+    public void goToHomePage()
+    {
+        getUtil().gotoPage("Main", "WebHome");
+    }
+
+    public void clickLogin()
+    {
+        toggleDrawer();
+        this.loginLink.click();
+    }
+
+    public void clickLoginBypass()
+    {
+        toggleDrawer();
+        this.bypassLogin.click();
+    }
+
+    public boolean canBypassLogin()
+    {
+        toggleDrawer();
+        return this.contentContainsElement(By.id("tmLogin-bypass"));
+    }
+
+    public boolean isSwitchUserDisplayed()
+    {
+        toggleDrawer();
+        List<WebElement> webElements = getDriver().findElements(By.id("tmLogout-xwiki-switch"));
+        return !webElements.isEmpty();
+    }
+
+    public WebElement getMicrosoftContainer()
```

Comment:
I suggest to move this to a Microsoft home page PO and specify on it that it's not our code and it can suffer changes that break the tests